docs/program/extend/modular-resources/examples/add-tflite-module.md

```diff
@@ -0,0 +1,232 @@
+---
+title: "Add a TensorFlow Lite Modular Service"
+linkTitle: "Add a TensorFlow Lite Modular Service"
+weight: 70
+type: "docs"
+description: "Add a custom MLModel modular-resource-based service which uses TensorFlow Lite to classify audio samples."
+tags: ["ml", "model training", "services"]
+# SMEs: Andrew Morrow
+---
+
+Viam provides an example [custom module](extend/modular-resources) written in C++ that extends the [ML model](/services/ml/) service to support audio classification using TensorFlow Lite.
+
+The example files can be found in the [Viam C++ SDK](https://github.com/viamrobotics/viam-cpp-sdk):
+
+- [`example_mlmodelservice_tflite.cpp`](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/examples/modules/example_mlmodelservice_tflite.cpp) - a custom module that provides an example `MLModelService` instance which runs TensorFlow Lite models.
+- [`example_audio_classification_client.cpp`](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/examples/mlmodel/example_audio_classification_client.cpp) - an example client instance which generates audio samples and invokes the `example_mlmodelservice_tflite` custom module to classify those samples.
+
+This tutorial walks you through everything necessary to start using these example files with your robot, including building the C++ SDK, procuring the necessary support files, configuring your robot and installing `viam-server`, and running the compiled binary example.
+
+## Build the C++ SDK
+
+To build the Viam C++ SDK, you will need a macOS or Linux computer.
+Follow the instructions below for your platform:
+
+{{< tabs >}}
+{{% tab name="macOS" %}}
+
+Follow the [Viam C++ SDK build instructions](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md) to build the SDK on your macOS computer using the `brew` package manager.
+
+While your specific build steps may differ slightly, your installation should generally resemble the following:
+
+1. Install all listed dependencies to support the Viam C++ SDK:
+
+   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   brew install abseil cmake boost grpc protobuf xtensor pkg-config ninja buf
+   ```
+
+1. Clone the Viam C++ SDK to your computer:
+
+   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   mkdir -p ~/example_workspace
+   cd ~/example_workspace
+   git clone git@github.com:viamrobotics/viam-cpp-sdk.git
+   ```
+
+1. Create a <file>build</file> directory to house the build target:
+
+   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   cd viam-cpp-sdk/
+   mkdir build
+   cd build
+   ```
+
+1. The Viam C++ SDK requires a specific version of `openssl` in order to build successfully.
```

Comment:
This isn't actually true as far as I know. I build the SDK against OpenSSL 3. What is required though is to point the C++ SDK build at the brew OpenSSL since it isn't "linked" by brew due to potential conflicts with the system version (which only sort of exists in a ghostly fashion these days). My `CMake` invocation has this in the environment to hook up to the brew OpenSSL in an architecture and ssl-version-independent fashion:

```
PKG_CONFIG_PATH=`brew --prefix`/opt/openssl/lib/pkgconfig
```

Reply:
Thank you!! Your version saves us a step! And you're correct regarding `openssl` version support, my mistake misreading BUILDING. Thank you!